pkg/operator/controller/canary/route_test.go

```diff
@@ -0,0 +1,96 @@
+package canary
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDesiredCanaryRoute(t *testing.T) {
+	deploymentRef := metav1.OwnerReference{
+		Name: "test",
+	}
+	service := desiredCanaryService(deploymentRef)
+	route := desiredCanaryRoute(service)
+
+	expectedRouteName := types.NamespacedName{
+		Namespace: "openshift-ingress-canary",
+		Name:      "ingress-canary-route",
+	}
+
+	if !cmp.Equal(route.Name, expectedRouteName.Name) {
+		t.Errorf("Expected route name to be %s, but got %s", expectedRouteName.Name, route.Name)
+	}
+
+	if !cmp.Equal(route.Namespace, expectedRouteName.Namespace) {
+		t.Errorf("Expected route namespace to be %s, but got %s", expectedRouteName.Namespace, route.Namespace)
+	}
+
+	expectedLabels := map[string]string{
+		"ingress.openshift.io/canary": "canary_controller",
+	}
+	if !cmp.Equal(route.Labels, expectedLabels) {
+		t.Errorf("Expected route labels to be %q, but got %q", expectedLabels, route.Labels)
+	}
+
+	routeToName := route.Spec.To.Name
+	if !cmp.Equal(routeToName, service.Name) {
+		t.Errorf("Expected route.Spec.To.Name to be %q, but got %q", service.Name, routeToName)
+	}
+
+	routeTarget := route.Spec.Port.TargetPort
+	validTarget := false
+	for _, port := range service.Spec.Ports {
+		if cmp.Equal(routeTarget, port.TargetPort) {
+			validTarget = true
+		}
+	}
+
+	if !validTarget {
+		t.Errorf("Expected route targetPort to be a port in the parameter service. Route targetPort not in service targetPort list")
```

Comment:
You can use `t.Error` here... or you could print `route.Spec.Port.TargetPort` and `service.Spec.Ports` in the error message.